app/src/main/kotlin/org/wordpress/aztec/demo/MainActivity.kt

```diff
@@ -178,23 +179,29 @@ class MainActivity : AppCompatActivity(),
             }
         }
 
-        aztec.setOverlay(predicate, 0, ColorDrawable(0x80000000.toInt()), Gravity.FILL, attrs)
-        val progressDrawable = resources.getDrawable(android.R.drawable.progress_horizontal)
+        aztec.setOverlay(predicate, 0, ColorDrawable(0x80000000.toInt()), Gravity.FILL)
+        aztec.updateElementAttributes(predicate, attrs)
```

Comment:
The method's name is `setOverlay()` yet it performed attribute updates as well, which was a hidden side effect that the caller might not be aware of just by looking at it (I know I wasn't). By the Single Responsibility Principle `setOverlay()` shoud set overlays, IMO. Also, in wpandroid there are cases where you need to apply several overlays and each call sets same attributes over and over.